### PR TITLE
Exposing ServerChannel

### LINKED ISSUE #267: Move `withServerChannel` to `freestyle.rpc.testing.servers`
The current implementation of `ServerChannel`, and in particular its feature `withServerChannel` for executing operations from RPC services in memory, needs to be moved to `freestyle.rpc.testing.servers`, in order to provide the capability of testing multiple RPC services in parallel without collisions.

### PR DESCRIPTION
**Pull request by rafaparadela**
This PR resolves #267 

In summary:

- It moves the implementation of `ServerChannel` to the `servers` object, in order to expose `withServerChannel` for testing purpose.
- It adapts this minor change in `RPCTests`, which is the only test suite that currently uses this feature.
- It adds a brief outline about how to test RPC services in memory in docs.

### CODE REVIEW

**Comment by L-Lavigne** [comment]
Looks good, could we add some unit tests similar to the code sample in the documentation? We should also make sure to include a bidirectional streaming test. I had issues with such a test intermittently deadlocking in a previous implementation of `ServerChannel`, which I worked around by removing the `directExecutor` here: https://github.com/frees-io/freestyle-rpc/blob/808c243600ad6c1a18d285a80e8123b0531865ec/modules/testing/src/test/scala/ServerChannel.scala#L60

**Comment by rafaparadela** [comment]
@L-Lavigne Unit test has been added in Docs to prove how to use `withServerChannel` and also, `avro/RPCTests.scala` is using `withServerChannel` from the new location. I didn't add any new functionality, I just moved it from one location to another.
I guess that if more tests are needed, these can be added progressively in future PRs.

File: docs/src/main/tut/patterns.md [hunk 1]
```diff
@@ -205,6 +205,49 @@ object RPCServer {
...
+
+  test("Get a valid response when a proper request is passed") {
+    sayHelloTest(requestGen, HelloResponse(reply = "Good bye!"))
+  }
+
+}
+
+```
+
+Passing the test:
```
[the remark this answers is not in the document]
Comment (rafaparadela) [reply docs/src/main/tut/patterns.md]: Thanks @L-Lavigne 
Fixed at https://github.com/frees-io/freestyle-rpc/pull/268/commits/db7ba6f9f8e6f88fb1cf515da36f15a325a6ad2c

File: docs/src/main/tut/patterns.md [hunk 2]
```diff
@@ -205,6 +205,49 @@ object RPCServer {
 
 Fortunately, once all the runtime requirements are in place (**`import gserver.implicits._`**), we only have to write the previous piece of code, which primarily, should be the same in all cases (except if your target Monad is different from `cats.effects.IO`).
 
+### Service testing
+
+Thanks to `withServerChannel` from the package `freestyle.rpc.testing.servers`, you will be able to run in-memory instances of the server, which is very convenient for testing purpose. Below, a very simple property-based test for proving `Greeter.sayHello`:
```
[the remark this answers is not in the document]
Comment (rafaparadela) [reply docs/src/main/tut/patterns.md]: Thanks @L-Lavigne 
Fixed at https://github.com/frees-io/freestyle-rpc/pull/268/commits/db7ba6f9f8e6f88fb1cf515da36f15a325a6ad2c

**Pull request merged**